cmd/billing/billing.go

```diff
@@ -0,0 +1,197 @@
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		usageData := resp.GetData()
+
+		switch strings.ToLower(format) {
+		case "csv":
+			if err := outputCSV(usageData, "usage.csv"); err != nil {
+				return
+			}
+		case "json":
+			if err := outputJSON(usageData, "usage.json"); err != nil {
+				return
+			}
+		default:
+			logrus.Warnf("Unsupported format: %s. Defaulting to CSV.", format)
+			if err := outputCSV(usageData, "usage.csv"); err != nil {
+				return
+			}
+		}
+	},
+}
+
+func getDimensionNames(resp ybmclient.BillingUsageData) []string {
+	var dimensionNames []string
+
+	if len(resp.GetDimensions()) > 0 {
+		for _, dimension := range resp.GetDimensions() {
+			dimensionNames = append(dimensionNames, string(dimension.Name))
+		}
+	}
+
+	return dimensionNames
+}
+
+func outputCSV(resp ybmclient.BillingUsageData, filename string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+
+	dimensionHeaders := []string{"Date"}
+	for _, dimension := range resp.GetDimensions() {
+		dimensionHeaders = append(dimensionHeaders, string(dimension.GetName())+"_Daily", string(dimension.GetName())+"_Cumulative")
+	}
+
+	err = writer.Write(dimensionHeaders)
+	if err != nil {
+		return err
+	}
+
+	for _, dataPoint := range resp.GetDimensions()[0].DataPoints {
+		row := []string{dataPoint.GetStartDate()}
+
+		for _, dimension := range resp.GetDimensions() {
+			var found bool
+			for _, point := range dimension.DataPoints {
+				if point.StartDate == dataPoint.StartDate {
+					row = append(row, fmt.Sprintf("%f", *point.DailyUsage))
+					row = append(row, fmt.Sprintf("%f", *point.CumulativeUsage))
+					found = true
+					break
+				}
+			}
+
+			// If no data found for the current dimension, add placeholders
+			if !found {
+				row = append(row, "0.000000", "0.000000")
+			}
+		}
+
+		err := writer.Write(row)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Printf("CSV data written to %s\n", filename)
+	return nil
+}
+
+func outputJSON(data interface{}, filename string) error {
+	jsonData, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal data to JSON: %v", err)
+	}
+
+	file, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("failed to create or open JSON file: %v", err)
+	}
+	defer file.Close()
+
+	_, err = file.Write(jsonData)
+	if err != nil {
+		return fmt.Errorf("failed to write JSON data to file: %v", err)
+	}
+
+	fmt.Printf("JSON data written to %s\n", filename)
+	return nil
+}
+
+func init() {
+	BillingCmd.AddCommand(listUsageCmd)
+	listUsageCmd.Flags().String("start", "", "[REQUIRED] Start date in yyyy-MM-dd format (e.g., '2023-09-01').")
+	listUsageCmd.Flags().String("end", "", "[REQUIRED] End date in yyyy-MM-dd format (e.g., '2023-09-30').")
+	listUsageCmd.Flags().StringSliceP("clusters", "c", []string{}, "[OPTIONAL] List of clusters")
```

Comment:
I think description could be enhanced : 
List of cluster name ? separated by what? comma? semi colon?

Reply:
Done